lib/plugins/aws/deploy/lib/splitStack.js

```diff
@@ -0,0 +1,212 @@
+'use strict';
+
+const path = require('path');
+const _ = require('lodash');
+const BbPromise = require('bluebird');
+const traverse = require('traverse');
+const DepGraph = require('dependency-graph').DepGraph;
+
+module.exports = {
+  splitStack() {
+    if (!this.serverless.service.provider.useStackSplitting) return BbPromise.resolve();
+
+    return BbPromise.bind(this)
+      .then(this.createDepGraph)
+      .then(this.generateStacks)
+      .then(this.uploadStackFiles)
+      .then(this.updateCompiledCloudFormationTemplate)
+      .then(this.writeStacksToDisk);
```

Comment:
For the new package/deploy semantics, it should be done in the deployment step, not the package step. Otherwise the deploy command would have to have too much knowledge about the stack splitting itself.
This has to be refactored anyway as soon as the new commands are released to hook the new command hooks.
Imo this should be discussed at a later time.